src/components/Span/SpanDetailsPanel.tsx

```diff
@@ -61,7 +57,11 @@ export const SpanDetailPanel = ({
         <div>
           <strong>Duration:</strong> <pre>{span.endTimeUnixNano - span.startTimeUnixNano}ms</pre>
         </div>
-        {JSON.stringify(result.data)}
+        {result.data && Object.keys(result.data).length > 0 && (
```

Comment:
This was always meant as a temporary measure.
Showing JSON is not the goal here; we should stick to how the detail pane in a regular Grafana trace look.

Reply:
Thanks, the important thing here will be to have a good format for each kind of value type.